src/Broadway/EventStore/MongoDBEventStore.php

```diff
@@ -0,0 +1,371 @@
+     */
+    public function configureCollection()
+    {
+        $this->eventCollection->ensureIndex(array(
+            self::FIELD_STREAM_ID => MongoCollection::ASCENDING,
+            self::FIELD_PLAYHEAD  => MongoCollection::ASCENDING,
+        ), array(
+            'unique'     => true,
+            'background' => true,
+        ));
+    }
+
+    /**
+     * @param DomainEventStreamInterface|DomainMessage[] $eventStream
+     * @param string                                     $aggregateId
+     *
+     * @return array
+     */
+    private function extractEvents(DomainEventStreamInterface $eventStream, $aggregateId)
+    {
+        $events = array();
+
+        foreach ($eventStream as $message) {
+            $this->assertIdentifier($aggregateId, $message->getId());
+
+            $events[] = $this->serializeEvent($message);
+        }
+
+        return $events;
+    }
+
+    /**
+     * @param string $expected
+     * @param string $actual
+     */
+    private function assertIdentifier($expected, $actual)
+    {
+        if (0 != strcmp($expected, $actual)) {
+            throw new InvalidIdentifierException(sprintf('Expecting %s, got %s identifier.', $expected, $actual));
+        }
+    }
+
+    /**
+     * @param DomainMessage $message
+     *
+     * @return array
+     */
+    private function serializeEvent(DomainMessage $message)
+    {
+        $data = array(
+            self::FIELD_STREAM_ID   => (string) $message->getId(),
+            self::FIELD_PLAYHEAD    => $message->getPlayhead(),
+            self::FIELD_METADATA    => $this->metadataSerializer->serialize($message->getMetadata()),
+            self::FIELD_PAYLOAD     => $this->payloadSerializer->serialize($message->getPayload()),
+            self::FIELD_RECORDED_ON => $message->getRecordedOn()->toString(),
+            self::FIELD_TYPE        => $message->getType(),
+        );
+
+        // save space
+        if (empty($data[self::FIELD_METADATA])) {
+            unset($data[self::FIELD_METADATA]);
+        }
+
+        return $data;
+    }
+
+    /**
+     * @param array $row
+     *
+     * @return DomainMessage
+     */
+    private function deserializeEvent(array $row)
+    {
+        if (empty($row[self::FIELD_METADATA])) {
+            $metadata = new Metadata();
+        } else {
+            $metadata = $this->metadataSerializer->deserialize($row[self::FIELD_METADATA]);
+        }
+
+        return new DomainMessage(
+            $row[self::FIELD_STREAM_ID],
+            $row[self::FIELD_PLAYHEAD],
+            $metadata,
+            $this->payloadSerializer->deserialize($row[self::FIELD_PAYLOAD]),
+            DateTime::fromString($row[self::FIELD_RECORDED_ON])
+        );
+    }
+
+    /**
+     * @param array $event
+     */
+    private function appendEvent(array $event)
+    {
+        $this->eventCollection->insert($event, array(
+            'safe' => true,
```

Comment:
Acknowledged writes have been the default since driver version 1.3, so this shouldn't be necessary. If you did want to specify the option, `['w' => 1]` would be the equivalent option to use (`safe` is deprecated).

I'm not sure how important the data being logged by this service is, but you may want `['w' => 'majority']` when working with a replica set (in that case, it'd probably be best to make the write concern a configurable option).